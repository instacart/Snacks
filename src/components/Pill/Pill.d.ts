import * as React from 'react'
import { Interpolation } from '@emotion/core'
import { WithThemeInjectedProps, ApplyWithTheme } from '../../styles/themer/withTheme'
import { ElementAttributes } from '../..'
import { Theme } from '../../styles/themer/utils'

export interface PillProps extends WithThemeInjectedProps {
  /** Color of the pill. can be hex value or themer color key as string */
  color?: keyof Theme['colors'] | string

  /** The pill's text content. */
  children: React.ReactNode

  className?: string

  /**
   * Optional style overrides merged into emotion css
   *
   * @deprecated
   * This prop exists for backwards compatibility and will be
   * removed in a future version
   */
  style?: Interpolation

  /** Any addional props. */
  elementAttributes?: ElementAttributes<React.ComponentPropsWithoutRef<'div'>>
}

declare const Pill: ApplyWithTheme<React.ComponentType<PillProps>>

export default Pill
