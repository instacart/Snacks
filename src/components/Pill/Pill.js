import React from 'react'
import spacing from '../../styles/spacing'
import PropTypes from 'prop-types'
import Radium from 'radium'
import withTheme from '../../styles/themer/withTheme'

const getStyles = props => {
  const { color: colorProp, snacksTheme } = props
  const color = snacksTheme.colors[colorProp] || colorProp

  return {
    backgroundColor: color,
    color: 'white',
    borderRadius: spacing.SM,
    paddingTop: 0,
    paddingBottom: 0,
    ...spacing.PADDING_X_XS,
    ...spacing.MARGIN_XS,
    textAlign: 'center',
    display: 'inline-block',
    width: 'auto',
  }
}

const Pill = props => {
  const styles = getStyles(props)

  return (
    <div
      style={[styles, props.style]}
      {...props.elementAttributes}
    >
      {props.children}
    </div>
  )
}

Pill.propTypes = {
  /** Color of the pill. can be hex value or themer color key as string */
  color: PropTypes.string,

  /** The pill's text content. */
  children: PropTypes.node.isRequired,

  /** Optional styles. */
  style: PropTypes.object,

  /** Any addional props. */
  elementAttributes: PropTypes.object
}

Pill.defaultProps = {
  color: '#CC0033'
}

export default withTheme(Radium(Pill))
