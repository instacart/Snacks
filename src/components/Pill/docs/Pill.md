### Pill example:

```jsx static
const defaultTheme = require('../../../styles/themer/utils').defaultTheme
const colors = require('../../../styles/colors').default

const internalTrackStyles = {
  RightArrow: {
    backgroundColor: 'blue',
  }
}

const styles = {
  padding: '8px 20px',
  marginRight: '5px',
  fontSize: '24px',
  borderRadius: '4px',
  backgroundColor: '#eee',
  color: '#43B02A',
  top: '8px',
}

<div>
  <h3>Passing in theme color keys </h3>
  <div style={{ height: '56px' }}>
    <ScrollTrack styles={internalTrackStyles}>
    <div style={{ padding: '8px 0' }}>
      {Object.keys(defaultTheme.colors).map(color => (
        <Pill color={color} style={{margin: '0.5rem'}}>{color}</Pill>
      ))}
      </div>
    </ScrollTrack>
  </div>

  <h3>Passing in Hex values</h3>
  <div style={{ height: '56px' }}>
    <ScrollTrack styles={internalTrackStyles}>
    <div style={{ padding: '8px 0' }}>
      {Object.values(colors).map(color => (
        <Pill color={color} style={{margin: '0.5rem'}}>{color}</Pill>
      ))}
      </div>
    </ScrollTrack>
  </div>
</div>
```
