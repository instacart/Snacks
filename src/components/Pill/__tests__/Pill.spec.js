import React from 'react'
import renderer from 'react-test-renderer'
import Pill from '../Pill'

describe('Pill', () => {
  it('renders without throwing', () => {
    const tree = renderer.create(<Pill>HI</Pill>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when style overrides are provided', () => {
    const tree = renderer.create(<Pill style={{ textDecoration: 'underline' }}>HI</Pill>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('applies the elementAttributes prop correctly', () => {
    const tree = renderer
      .create(<Pill elementAttributes={{ 'aria-label': 'foo' }}>HI</Pill>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
