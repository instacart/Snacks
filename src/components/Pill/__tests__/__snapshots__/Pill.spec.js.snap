// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill applies the elementAttributes prop correctly 1`] = `
<div
  aria-label="foo"
  data-radium={true}
  style={
    Object {
      "backgroundColor": "#CC0033",
      "borderRadius": 16,
      "color": "white",
      "display": "inline-block",
      "marginBottom": 8,
      "marginLeft": 8,
      "marginRight": 8,
      "marginTop": 8,
      "paddingBottom": 0,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 0,
      "textAlign": "center",
      "width": "auto",
    }
  }
>
  HI
</div>
`;

exports[`Pill renders correctly when style overrides are provided 1`] = `
<div
  data-radium={true}
  style={
    Object {
      "backgroundColor": "#CC0033",
      "borderRadius": 16,
      "color": "white",
      "display": "inline-block",
      "marginBottom": 8,
      "marginLeft": 8,
      "marginRight": 8,
      "marginTop": 8,
      "paddingBottom": 0,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 0,
      "textAlign": "center",
      "textDecoration": "underline",
      "width": "auto",
    }
  }
>
  HI
</div>
`;

exports[`Pill renders without throwing 1`] = `
<div
  data-radium={true}
  style={
    Object {
      "backgroundColor": "#CC0033",
      "borderRadius": 16,
      "color": "white",
      "display": "inline-block",
      "marginBottom": 8,
      "marginLeft": 8,
      "marginRight": 8,
      "marginTop": 8,
      "paddingBottom": 0,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 0,
      "textAlign": "center",
      "width": "auto",
    }
  }
>
  HI
</div>
`;
